erpnext/hr/doctype/salary_slip/salary_slip.js

```diff
@@ -83,6 +83,10 @@ frappe.ui.form.on("Salary Slip", {
 		var salary_detail_fields = ["formula", "abbr", "statistical_component", "variable_based_on_taxable_salary"];
 		cur_frm.fields_dict['earnings'].grid.set_column_disp(salary_detail_fields,false);
 		cur_frm.fields_dict['deductions'].grid.set_column_disp(salary_detail_fields,false);
+
+		frm.add_custom_button(__('Update Salary Slip'), function () {
```

Comment:
Make sure this button does not show after Salary Slips are submitted... add the button along with where "Submit Salary Slips" button is